### ISSUE TITLE
can't modify frozen string

### ISSUE THREAD

**Issue by gaiottino** [bcurren/freshbooks.rb#6]
Trying to use freshbooks.rb in a Rails 3 project running on Ruby 1.9.2 and ran into a problem. I got the error below. The error is thrown as soon as I try to run

FreshBooks::Client.list

or

FreshBooks::Client.get(id)

in the rails console. If I run the same code under Ruby 1.8.7 I don't get the error.

FreshBooks::ParseError: Original Error: can't modify frozen string
XML: <client>
      <client_id>26</client_id>
      <username>danielgaiottino14</username>
      <first_name>Daniel</first_name>
      <last_name>Gaiottino</last_name>
      <organization>Burt</organization>
      <email>daniel@gaiottino.com</email>
      <work_phone/>
      <home_phone/>
      <mobile/>
      <fax/>
      <vat_name/>
      <vat_number/>
      <p_street1/>
      <p_street2/>
      <p_city/>
      <p_state/>
      <p_country/>
      <p_code/>
      <s_street1/>
      <s_street2/>
      <s_city/>
      <s_state/>
      <s_country/>
      <s_code/>
      <notes/>
      <language>en</language>
      <currency_code/>
      <credit currency='SEK' deprecated='true'>0</credit>
      <credits>
        <credit currency='SEK'>0</credit>
      </credits>
      <url deprecated='true'>https://burtcorp.freshbooks.com/view/3LyJG2G8EcmiBBKg</url>
      <auth_url deprecated='true'>https://burtcorp.freshbooks.com/clients/26</auth_url>
      <links>
        <client_view>https://burtcorp.freshbooks.com/view/3LyJG2G8EcmiBBKg</client_view>
        <view>https://burtcorp.freshbooks.com/clients/26</view>
        <statement>https://burtcorp.freshbooks.com/view/Nnck2iLJKXuUTwe</statement>
      </links>
      <updated>2010-11-01 17:29:11</updated>
      <folder>active</folder>
    </client>
Message: FreshBooks::ParseError

```
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:31:in `rescue in new_from_xml'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:18:in `new_from_xml'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:129:in `block (2 levels) in api_list_action'
from /Users/daniel/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/rexml/element.rb:906:in `block in each'
from /Users/daniel/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/rexml/xpath.rb:64:in `each'
from /Users/daniel/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/rexml/xpath.rb:64:in `each'
from /Users/daniel/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/rexml/element.rb:906:in `each'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:129:in `map'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:129:in `block in api_list_action'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/list_proxy.rb:35:in `call'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/list_proxy.rb:35:in `move_to_page'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/list_proxy.rb:7:in `initialize'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:136:in `new'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:136:in `api_list_action'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/bcurren-freshbooks.rb-3.0.12/lib/freshbooks/base.rb:98:in `block (2 levels) in actions'
from (irb):1
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/railties-3.0.1/lib/rails/commands/console.rb:44:in `start'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/railties-3.0.1/lib/rails/commands/console.rb:8:in `start'
from /Users/daniel/.rvm/gems/ruby-1.9.2-p0@ka-ching/gems/railties-3.0.1/lib/rails/commands.rb:23:in `<top (required)>'
from script/rails:6:in `require'
from script/rails:6:in `<main>'
```

**Comment by bcurren** [comment]
This should have been resolved when I merged in 1.9.2 support. Let me know if you still see the issue.

**Comment by superchris** [comment]
Still seems to be broken just now when I tried it.  Was doing Invoice.list, if that matters
